main.go

```diff
@@ -0,0 +1,197 @@
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+	options.SetSort(bson.D{{"name", 1}}) // Sort by `name` field ascending
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		log.Fatal("Something went wrong trying to find the documents: ", err)
+	}
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		// declare a result BSON object
+		var result bson.M
+		err := cursor.Decode(&result)
+		// If there is a cursor.Decode error
+		if err != nil {
+			log.Fatal("cursor.Next() error:", err)
+		} else {
+			var name = result["name"].(string)
+			var ingredients = result["ingredients"]
+			var prepTimeInMinutes = result["prepTimeInMinutes"].(int32)
+
+			fmt.Println(name, "has the ingredients:", ingredients, "and takes", prepTimeInMinutes, "minutes to make")
+		}
+	}
+
+	// We can also find a single document. Let's find the first document
+	// that has the string "fried rice" in the name.
+	var result Recipe
+	var myFilter = bson.D{{"name", "fried rice"}}
+	e := collection.FindOne(context.TODO(), myFilter).Decode(&result)
+	if e != nil {
+		log.Fatal("Something went wrong trying to find one document:", e)
+	}
+	fmt.Println("found a document with the name fried rice", result)
+
+	/*
+	 * *** UPDATE A DOCUMENT ***
+	 *
+	 * You can update a single document or multiple documents in a single call.
+	 *
+	 * Here we update the prepTimeInMinutes value on the document we
+	 * just found.
+	 */
+
+	var updateDoc = bson.D{{"$set", bson.D{{"prepTimeInMinutes", 72}}}}
+	var myRes = collection.FindOneAndUpdate(ctx, myFilter, updateDoc, nil)
+	if myRes.Err() != nil {
+		log.Fatal("Something went wrong trying to update one document:", myRes.Err())
+	}
+	doc := bson.M{}
+	decodeErr := myRes.Decode(&doc)
+	if decodeErr != nil {
+		log.Fatal("Something went wrong trying to decode the document:", decodeErr)
+	}
+	fmt.Println("Here is the updated document:", doc)
+
+	/*      *** DELETE DOCUMENTS ***
+	 *
+	 *      As with other CRUD methods, you can delete a single document
+	 *      or all documents that match a specified filter. To delete all
+	 *      of the documents in a collection, pass an empty filter to
+	 *      the DeleteMany() method. In this example, we'll delete two of
+	 *      the recipes.
+	 */
+
+	deletedRecipeNameList := make([]string, 0)
+	deletedRecipeNameList = append(deletedRecipeNameList, "elotees")
+	deletedRecipeNameList = append(deletedRecipeNameList, "fried rice")
+
+	var deleteQuery = bson.M{"name": bson.M{"$in": deletedRecipeNameList}}
+	deleteResult, err := collection.DeleteMany(context.TODO(), deleteQuery)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("Deleted %v documents in the recipes collection\n", deleteResult.DeletedCount)
```

Comment:
Would it make sense to stay consistent by using either Println vs Printf when including one or more inline variable values? I prefer Printf since you're generally specifying the type while Println has to try to figure it out.

Reply:
switched to Println for consistency